cmake/modules/FindMusic.cmake

```diff
@@ -0,0 +1,12 @@
+#
+# Find the MUSIC installation.
+#
+
+find_path(MUSIC_INCDIR music.hh PATH_SUFFIXES include)
+find_path(MUSIC_LIBDIR libmusic.a PATH_SUFFIXES lib)
+find_library(MUSIC_LIBRARY libmusic.a PATHS ${MUSIC_LIBDIR})
```

Comment:
Nice. (I think). I believe this means that the music code appears in libnrniv.so .  Am I right also in thinking that music support can be enabled in wheels with no negative effects? (although to actually use it the user will have to install music as well.

Reply:
So, I realized that I should add a little extra explanation: The way you make NEURON aware of MUSIC with this build system extension is by adding a `-DMUSIC_ROOT=/path/to/music/prefix` or alternatively a `-DCMAKE_PREFIX=/path/to/music/prefix` to your `cmake` command.

What `find_path` does, is check a bunch of directories, including directories passed in the arguments above. It checks specifically for: `/path/to/music/prefix/include/music.hh` and `/path/to/music/prefix/lib/libmusic.a` . The `find_library` works similarly, but there I tell the function to first check the variable I've set in the call just above.

The wheels part I haven't tested, but if we enable MUSIC for building the wheel then we need at build time `libmusic.a`, against which we link. I'm not sure this will just work then at runtime in the wheel - we might need to do again the dynamic loading trick? @pramodk , you know this stuff better...

Reply:
Ok. At present, there is no provision for  MUSIC building libmusic.a with ```-fPIC``` except to hack config.status after running ./configure. I tried that and libmusic.a does end up inside  libnrniv.so. However I don't know if the final environment  with several simulators, some of which use libmusic.so, and NEURON which will be using the music objs in
libnrniv.so will nicely work together. At any rate, I expect if the user builds everything from sources that libnrniv.so should link to libmusic.so from the music installation, i.e. ldd libnrniv.so should list libmusic.so as one of its needed libraries.
Another issue hiding in the shadows is that if the NEURON cmake configure asks for ```-DNRN_ENABLE_MPI_DYNAMIC=ON```, and we add libmusic.a into libnrniv.so. Everything goes south because the libmusic.a stuff links to explicit MPI functions instead of wrappers for later dynamic loading of MPI.  So, I guess my feeling for this PR is to make sure we link to libmusic.so and forget about adding MUSIC support to our wheels.

Reply:
Oh very good point you're making on the dynamic MPI.
What do you think then of leaving MUSIC as feature that is only available in the source distribution. People who want to the MUSIC integration will have to build NRN from sources with MUSIC support enabled (after having obtained MUSIC, whichever way they prefer).
In that case I am going to switch everything to linking against the `.so`. Also, you can discard my comment further down 😄 .

Reply:
It would of course be good if MUSIC could be changed in such a way that you would find it easier to build a MUSIC-enabled NEURON wheel. (This would be solved through dynamic linking.) I'll complete the MPI interface conversion and release that. Afterwards, we could maybe discuss further modifications in a separate ticket? In any case, I agree with @nrnhines that it would be a bad idea to include pieces of MUSIC code in ordinary NEURON binaries.

Reply:
A MUSIC enabled NEURON wheel is definitely feasible (for a separate PR). If the user later installs a MUSIC wheel, or even just MUSIC from sources and we have a way for NEURON to know where libmusic.so is, then  NEURON will be able to participate in MUSIC simulations. The strategy for this is the same as for dynamic MPI. No change to MUSIC is required for this. (Ironically, perhaps not even changing MUSIC to not use the MPI C++ API :))

Reply:
See #2096